packages/scxa-faceted-search-results-v5/src/FilterSidebar.js

```diff
@@ -0,0 +1,140 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import CheckboxFacetGroup from './facetgroups/CheckboxFacetGroup'
+import {withFetchLoader} from "@ebi-gene-expression-group/atlas-react-fetch-loader";
+import MultiselectDropdownFacetGroup from "./facetgroups/MultiselectDropdownFacetGroup";
+
+const CheckboxFacetGroupWithFetchLoader = withFetchLoader(CheckboxFacetGroup)
+const MultiselectDropdownFacetGroupWithFetchLoader = withFetchLoader(MultiselectDropdownFacetGroup)
+
+class FilterSidebar extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      queryParams: props.queryParams,
+      changedFacetGroup: props.changedFacetGroup
+    }
+  }
+
+  getQueryParamsByFacetGroupsAsString(queryParams, facetGroupName) {
+    return queryParams[facetGroupName] === undefined ? [] : queryParams[facetGroupName]
+  }
+
+  initialiseFacetGroups(checkboxFacetGroups, multiSelectDropdownFacetGroups, changedFacetGroup, facetsOfChangedGroup) {
+    checkboxFacetGroups = checkboxFacetGroups
+      .map(facetGroup => ({...facetGroup, refreshGroup: true}))
+
+    checkboxFacetGroups
+      .filter(facetGroup => facetGroup.queryParamName === changedFacetGroup)
+      .map(facetGroup => {
+        facetGroup.facets = facetsOfChangedGroup
+        facetGroup.refreshGroup = false
+      })
+
+    multiSelectDropdownFacetGroups = multiSelectDropdownFacetGroups
+      .map(facetGroup => ({...facetGroup, refreshGroup: true}))
+
+    multiSelectDropdownFacetGroups
+      .filter(facetGroup => facetGroup.queryParamName === changedFacetGroup)
+      .map(facetGroup => {
+        facetGroup.facets = facetsOfChangedGroup
+        facetGroup.refreshGroup = false
+      })
+
+    return {
+      checkboxFacetGroups: checkboxFacetGroups,
+      multiSelectDropdownFacetGroups: multiSelectDropdownFacetGroups
+    }
+  }
+
+  render() {
+    const { host, onChange } = this.props
+    const queryParamsObject = this.state.queryParams
```

Comment:
I’ve seen `queryParams` renamed to `queryStringObject`, `queryParamsObject` (here) or just `queryParams`. I would settle on a single name. What’s wrong with just `queryParams`?